### PR TITLE
[RFC] Fix oldtest on Windows

### PR DESCRIPTION
**Pull request by erw7**
This is an attempt to fix oldtest on Windows.

### CODE REVIEW

**Comment by erw7** [comment]
The test52 will succeed in the next patch, but I think it is better to remove it.

```diff
--- a/src/nvim/testdir/test52.in
+++ b/src/nvim/testdir/test52.in
@@ -7,27 +7,27 @@ STARTTEST
 :" write tests:
 :" combine three values for 'encoding' with three values for 'fileencoding'
 :" also write files for read tests
+:e ++enc=utf-8
 /^1
-:set encoding=utf-8
 :.w! ++enc=utf-8 test.out
 :.w ++enc=cp1251 >>test.out
 :.w ++enc=cp866 >>test.out
 :.w! ++enc=utf-8 Xutf8
 /^2
-:set encoding=cp1251
+:e ++enc=cp1251
 :.w ++enc=utf-8 >>test.out
 :.w ++enc=cp1251 >>test.out
 :.w ++enc=cp866 >>test.out
 :.w! ++enc=cp1251 Xcp1251
 /^3
-:set encoding=cp866
+:e ++enc=cp866
 :.w ++enc=utf-8 >>test.out
 :.w ++enc=cp1251 >>test.out
 :.w ++enc=cp866 >>test.out
 :.w! ++enc=cp866 Xcp866
 :"
 :" read three 'fileencoding's with utf-8 'encoding'
-:set encoding=utf-8 fencs=utf-8,cp1251
+:set fencs=utf-8,cp1251
 :e Xutf8
 :.w ++enc=utf-8 >>test.out
 :e Xcp1251
@@ -37,7 +37,7 @@ STARTTEST
 :.w ++enc=utf-8 >>test.out
 :"
 :" read three 'fileencoding's with cp1251 'encoding'
-:set encoding=utf-8 fencs=utf-8,cp1251
+:set fencs=utf-8,cp1251
 :e Xutf8
 :.w ++enc=cp1251 >>test.out
 :e Xcp1251
@@ -47,7 +47,7 @@ STARTTEST
 :.w ++enc=cp1251 >>test.out
 :"
 :" read three 'fileencoding's with cp866 'encoding'
-:set encoding=cp866 fencs=utf-8,cp1251
+:set fencs=utf-8,cp1251
 :e Xutf8
 :.w ++enc=cp866 >>test.out
 :e Xcp1251
```

**Comment by justinmk** [comment]
> The test52 will succeed in the next patch, but I think it is better to remove it.

👍  That test was missed when we removed `'encoding'` because it only runs on Windows.

> When using `set binary`, change to use `++bin` argument because it is not
`nobinary` when returning to the original file with `:e #` on line 18 of
test42.in.

Maybe `set cpo+=S` is needed? https://github.com/neovim/neovim/issues/10730#issuecomment-519885658

## Status

Waiting on https://github.com/neovim/neovim/pull/10650. cc @blueyed

**Comment by erw7** [comment]
> Maybe set cpo+=S is needed?

After all, `set cpo+=S` was necessary. With #10662 and this pull request, all of oldtest succeeded on Windows :tada:.

**Comment by justinmk** [comment]
> should I set it in `setup.vim`?

Seems like a good idea. I wonder why Vim's tests are running with a different 'cpo'? 

> With #10662 and this pull request, all of oldtest succeeded on Windows 🎉.

Beautiful! Thanks for figuring this out.

**Comment by erw7** [comment]
> I wonder why Vim's tests are running with a different 'cpo'?

The following description exists in Vim: starting.txt.

> Using the "-u" argument with another argument than DEFAULTS has the side effect that the 'compatible' option will be on by default.  This can have unexpected effects.

Tests are run like `vim -u dos.vim` or `vim -u unix.vim`. Also, `set nocompatible` is not done in any of `dos.vim`, `unix.vim` and `setup.vim`. From the above, vim tests are run with `compatible`, therefore, the `cpo` during test is different between neovim and vim.

**Comment by erw7** [comment]
>> should I set it in setup.vim?

> Seems like a good idea.

So after #10650 is merged, change it that way.

**Comment by erw7** [comment]
#10764 ? After that, some new Windows oldtests seem to fail. Is there a missing patch?

#10764 
https://ci.appveyor.com/project/neovim/neovim/builds/26676309/job/g7kgwc3kjx45s9ie#L16117

before
https://ci.appveyor.com/project/neovim/neovim/builds/26675459/job/1an3i4gd6alwsxrc#L16075

File: src/nvim/testdir/shared.vim [hunk 1]
```diff
@@ -202,9 +202,17 @@ endfunc
 func GetVimProg()
   if empty($NVIM_TEST_ARG0)
     " Assume the script was sourced instead of running "make".
-    return '../../../build/bin/nvim'
+    if has('win32')
+      return '..\\..\\..\\build\\bin\\nvim.exe'
+    else
+      return '../../../build/bin/nvim'
+    endif
```
Comment (justinmk) [inline src/nvim/testdir/shared.vim]: Let's just use `v:progpath` here. Vim's reason for not using it is weak.

**Pull request merged**